applications/flash_attention_v2/collective/xe_flash_attn_mma.hpp

```diff
@@ -188,136 +212,114 @@ struct CollectiveMmaAttention<MainloopIntelPVC<Stages>, TileShape_, ElementQ_, S
+    TiledMmaQVO tiled_mma_q;
     // To make all threads in a warp have the same global tensors pass in the index of thread 0 in each warp
     auto sg = syclcompat::get_nd_item<1>().get_sub_group();
     auto first_thread_in_sg_idx = sg.get_group_id()[0] * DispatchPolicy::SubgroupSize;
-    auto thread_mma = tiled_mma.get_slice(first_thread_in_sg_idx);
-    // For partitioning tensor B we also want all subgroups in a row to load the same data.
-    // We have the same TiledMMA for both MMAs in flash attention, but the first one should have different strides.
-    // For now we have this hacky workaround
-    // the K matrix consumed from both dimention, hence starts from 0.  
-    auto thread_mma_b = tiled_mma.get_slice(0);
+    auto thread_mma_k = tiled_mma_k.get_slice(first_thread_in_sg_idx);
+    auto thread_mma_q = tiled_mma_q.get_slice(first_thread_in_sg_idx);
 
     // Partition
-    Tensor tCgA = thread_mma.partition_A(gA);
-    Tensor tCgB = thread_mma_b.partition_B(gB);
+    Tensor tCgQ = thread_mma_q.partition_A(gQ);
+    Tensor tCgK = thread_mma_k.partition_B(gK);
 
     // Create fragments
     // TODO(Codeplay): fix this, this is probably not general
-    Tensor tCrA = make_tensor<ElementQ>(make_fragment_layout(params.gmem_tiled_copy_q, take<0,3>(tCgA.shape())));
-    Tensor tCrB = make_tensor<ElementK>(make_fragment_layout(params.gmem_tiled_copy_k, take<0,3>(tCgB.shape())));
+    Tensor tCrQ = make_tensor<ElementQ>(make_fragment_layout(params.gmem_tiled_copy_q, take<0,3>(tCgQ.shape())));
+    Tensor tCrK = make_tensor<ElementK>(make_fragment_layout(params.gmem_tiled_copy_k, take<0,3>(tCgK.shape())));
     
     // Retile registers for copies
-    Tensor tArA = thr_copy_A.retile_D(tCrA);
-    Tensor tBrB = thr_copy_B.retile_D(tCrB);
+    Tensor tQrQ = thr_copy_Q.retile_D(tCrQ);
+    Tensor tKrK = thr_copy_K.retile_D(tCrK);
 
     // Retile global tile for copies
-    Tensor tAgA = thr_copy_A.retile_S(tCgA);
-    Tensor tBgB = thr_copy_B.retile_S(tCgB);
+    Tensor tQgQ = thr_copy_Q.retile_S(tCgQ);
+    Tensor tKgK = thr_copy_K.retile_S(tCgK);
 
 #if CUTLASS_ENABLE_DEBUG_PRINTS
-    if (thread(LOG_THREAD, LOG_GROUP)) {
-      print("======================= A: \n");
-      print("  gA : ");
-      print(gA);
-      print("\n");
-      print("tCgA : ");
-      print(tCgA);
-      print("\n");
-      print("  tCrA : ");
-      print(tCrA);
-      print("\n");
-
-      print("=====================  B :\n");
-      print("  gB : ");
-      print(gB);
-      print("\n");
-      print("tCrB_copy_view : ");
-      print(tCrB_copy_view);
-      print("\n");
-      print("  tCrB : ");
-      print(tCrB);
-      print("\n");
-
-      print("=====================  Config: \n");
-      print("  threads per workgroup : ");
-      print(MaxThreadsPerBlock);
-      print("\n");
-      print("  SubgroupTileShape : ");
-      print(SubgroupTileShape{});
-      print("\n");
-    }
+#define PRINT(x) print(#x ": "); print(x); print("\n");
```

Comment:
Can we move this define in some existing helper file like debug, or device_dump since it is used by several mma files

Reply:
Yeah, but let's do that in a separate PR so we can cover all the relevant pieces of code in a single go.